tests/strategies/test_bagua_strategy.py

```diff
@@ -0,0 +1,121 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import mock
+
+import pytest
+import torch
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.strategies import BaguaStrategy
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.runif import RunIf
+
+
+class BoringModel4QAdam(BoringModel):
+    def configure_optimizers(self):
+        from bagua.torch_api.algorithms.q_adam import QAdamOptimizer
+
+        optimizer = QAdamOptimizer(self.layer.parameters(), lr=0.05, warmup_steps=20)
+        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
+        return [optimizer], [lr_scheduler]
+
+
+@RunIf(skip_windows=True, bagua=True, min_gpus=1)
```

Comment:
Does Bagua support MacOs?

Reply:
no it does not. 
it seems the bagua package is also not installable on macos so the guard bagua=True here seems to be enough, but we could also add a skip condition to the test here. As you wish.